src/Stratis.Bitcoin.Features.RPC/RpcSettings.cs

```diff
@@ -32,74 +32,118 @@ public class RpcSettings
         public int RPCPort { get; set; }
 
         /// <summary>Default bindings from config.</summary>
-        public List<IPEndPoint> DefaultBindings { get; set; }
+        public List<IPEndPoint> DefaultBindings => DefaultBinding(this.AllowIp.Any(), this.RPCPort);
 
         /// <summary>List of network endpoints that the node will listen and provide RPC on.</summary>
         public List<IPEndPoint> Bind { get; set; }
 
         /// <summary>List of IP addresses that are allowed to connect to RPC interfaces.</summary>
         public List<IPAddressBlock> AllowIp { get; set; }
 
+        // TODO: Refactor static PrintHelp method.
+        private static RpcSettings settingsHelp;
+
+        private static List<IPEndPoint> DefaultBinding(bool haveAllowIps, int rpcPort)
+        {
+            var defaultBinding = new List<IPEndPoint>();
+
+            if (haveAllowIps)
+            {
+                defaultBinding.Add(new IPEndPoint(IPAddress.Parse("::"), rpcPort));
+                defaultBinding.Add(new IPEndPoint(IPAddress.Parse("0.0.0.0"), rpcPort));
+            }
+            else
+            {
+                defaultBinding.Add(new IPEndPoint(IPAddress.Parse("::1"), rpcPort));
+                defaultBinding.Add(new IPEndPoint(IPAddress.Parse("127.0.0.1"), rpcPort));
+            }
+
+            return defaultBinding;
+        }
+
         /// <summary>
         /// Initializes an instance of the object from the node configuration.
         /// </summary>
         /// <param name="nodeSettings">The node configuration.</param>
-        public RpcSettings(NodeSettings nodeSettings)
+        /// <param name="defaultArgs">Allows overriding the default values before config and command line are applied.</param>
+        public RpcSettings(NodeSettings nodeSettings, Action<RpcSettings> defaultArgs = null)
         {
             Guard.NotNull(nodeSettings, nameof(nodeSettings));
 
             this.logger = nodeSettings.LoggerFactory.CreateLogger(typeof(RpcSettings).FullName);
 
+            this.RPCPort = nodeSettings.Network.DefaultRPCPort;
             this.Bind = new List<IPEndPoint>();
-            this.DefaultBindings = new List<IPEndPoint>();
             this.AllowIp = new List<IPAddressBlock>();
+            this.Server = false;
+            this.RpcUser = null;
+            this.RpcPassword = null;
 
-            // Get values from config
+            defaultArgs?.Invoke(this);
+
+            // For printing help.
+            // TODO: Refactor.
+            settingsHelp = (RpcSettings)this.MemberwiseClone();
+
+            // Get values from config and command line.
             this.LoadSettingsFromConfig(nodeSettings);
 
             // Check validity of settings
             this.CheckConfigurationValidity(nodeSettings.Logger);
```

Comment:
Perhaps now that `CheckConfigurationValidity` is so thin (i.e. only checking the user and password) we can test the validity before actually loading the settings from config? It seems a bit backwards to first set all the values and then check if its valid?

Reply:
It seems perfectly valid to first set up the final values before checking the validity thereof.